iree/hal/driver_registry.c

```diff
@@ -60,6 +60,39 @@ iree_hal_driver_registry_default() {
   return &iree_hal_driver_registry_default_;
 }
 
+// Creates and initializes an empty device driver registry
+IREE_API_EXPORT iree_status_t IREE_API_CALL iree_hal_driver_registry_create(
+    iree_allocator_t allocator, iree_hal_driver_registry_t** out_registry) {
+  IREE_TRACE_ZONE_BEGIN(z0);
+
+  IREE_RETURN_IF_ERROR(iree_allocator_malloc(
```

Comment:
`IREE_END_ZONE_AND_RETURN_IF_ERROR` (otherwise the zone above won't be ended if the alloc fails)